src/System.Data.Common/src/System/Data/DbSchemaFactory.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+namespace System.Data.Common
+{
+    public abstract class DbSchemaFactory
+    {
+        abstract public List<DbColumn> GetColumnSchema(DbDataReader dataReader);
```

Comment:
Should this be `List<T>`? Has this been API reviewed?

The framework design guidelines say not to use `List<T>` in public APIs... Usually the recommendation is `Collection<T>` or `ReadOnlyCollection<T>` (or a subclass of one of those). In this case, it seems like you want some kind of read-only collection at the very least, because it doesn't seem like a caller of `GetColumnSchema` should be modifying the collection.

(Though, I am torn about this guideline as `List<T>` is more efficient than `Collection<T>` and `ReadOnlyCollection<T>`. Perhaps the guidelines should be reconsidered with an eye towards potentially allowing the use of immutable collections in APIs?)

cc: @terrajobst @KrzysztofCwalina